pylint/checkers/ellipsis_checker.py

```diff
@@ -0,0 +1,41 @@
+"""Ellipsis checker for Python code
+"""
+from astroid import nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+class EllipsisChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "unnecessary_ellipsis"
+    msgs = {
+        "W2301": (
+            "Unnecessary ellipsis constant",
+            "unnecessary-ellipsis",
+            "Used when the ellipsis constant is encountered and can be avoided.",
+        )
+    }
+
+    @check_messages("unnecessary-ellipsis")
+    def visit_const(self, node: nodes.Const) -> None:
+        """Check if the ellipsis constant is used unnecessarily"""
```

Comment:
Could you add an explanation of when an ellipsis is unnecessary? That would help potential contributors understand what these `if` statements are trying to determine.